akka-http-core/src/main/scala/akka/http/impl/util/Java6Compat.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.impl.util
+
+import java.lang.reflect.{ InvocationTargetException, Method }
+import javax.net.ssl.SSLParameters
+
+import scala.util.control.NonFatal
+
+/**
+ * INTERNAL API
+ *
+ * Enables accessing SslParameters even if compiled against Java 6.
+ */
+private[http] object Java6Compat {
```

Comment:
Mmh, we already have `EnhancedInetSocketAddress.getHostStringJava6Compatible`.

Would `EnhancedSSLParameters.setEndpointIdentificationAlgorithmJava8Only` be an option here?